js/simon.js

```diff
@@ -1,65 +1,53 @@
-var div1 = $(".div1" );
-var div2 = $(".div2" );
-var div3 = $(".div3" );
-var div4 = $(".div4" );
-
-var button1 = $(".button1");
-var button2 = $(".button2");
-var button3 = $(".button3");
-var button4 = $(".button4");
-
-div1.css("opacity", 1.0);
-div2.css("opacity", 0.0);
-div3.css("opacity", 0.0);
-div4.css("opacity", 0.0);
-
-var anim = false;
-var active = div1;
-var selected = button1;
-button1.addClass("menuClicked")
-
-button1.on("click", function(e) {
-	if (!anim && active != div1) {
-		selected.removeClass("menuClicked");
-		button1.addClass("menuClicked");
-		selected = button1;
-		swap(active, div1);
+(function(){
+	"use strcit";
+	
+	var div1 = $(".div1" );
+	var div2 = $(".div2" );
+	var div3 = $(".div3" );
+	var div4 = $(".div4" );
+	
+	var button1 = $(".button1");
+	var button2 = $(".button2");
+	var button3 = $(".button3");
+	var button4 = $(".button4");
+	
+	div1.css("display", '');
+	div2.css("display", 'none');
+	div3.css("display", 'none');
+	div4.css("display", 'none');
+	
+	var anim = false;
+	var active = div1;
+	var selected = button1;
+	button1.addClass("menuClicked")
+	
+	
+	var link = function(target_div, target_button){
+	  target_button.on('click', 
+			function(e) {
+				if (!anim && active != target_div) {
+					selected.removeClass("menuClicked");
+					target_button.addClass("menuClicked");
+					selected = target_button;
+					swap(active, target_div);
+				}
+			}	
+	  )
 	}
-});
-button2.on("click", function(e) {
-	if (!anim && active != div2) {
-		selected.removeClass("menuClicked");
-		button2.addClass("menuClicked");
-		selected = button2;
-		swap(active, div2);
+	
+	link(div1, button1);
+	link(div2, button2);
+	link(div3, button3);
+	link(div4, button4);
+	
+	var swap = function(div1, div2) {
+		anim = true;
+		div1.fadeOut( 200 , function() {
+			div2.fadeIn(200, function() {
+				active = div2;
+				anim = false;
+			}); 
+	
+		});
```

Comment:
Use `fadeIn/Out` which also toggle the `display:none` css property, that's what's forced you to use absolute positioning and all the complex height stuff I have partially removed above.  CAn be more simplified, but later. 
I also indented the file which make it looks like there is a lot of changes.